app/controllers/api/instances_controller.rb

```diff
@@ -140,8 +140,12 @@ def pause_instance(instance)
     end
 
     def validate_instance_for_action(instance, action)
-      validation = instance.send("validate_#{action}")
-      action_result(validation[:available], validation[:message].to_s)
+      if instance.respond_to?("supports_#{action}?")
+        action_result(instance.public_send("supports_#{action}?"), instance.unsupported_reason(action.to_sym))
```

Comment:
> Is there an existing API for features

@durandom created the SupportsFeatureMixin to move toward a standardized interface for declaring these things.  Previously it was a similar but less formalized interface around the `is_available?` method.  Work is on going to convert from one to the other, and this PR is one of those efforts, but in the middle you are going to have a bit if a hybrid.